src/pages/blog/post/der-komplette-guide-rund-um-sql-injections.mdx

```diff
@@ -0,0 +1,303 @@
+
+* **Error Based SQL-Injection**: Bei dieser Art von Angriff verursacht der Angreifer absichtlich einen Fehler im SQL-Code, um Informationen über die Struktur und den Inhalt der Datenbank zu erhalten. Diese Informationen können dann für weitere Angriffe verwendet werden.
+
+* **Blind SQL-Injection**: Bei dieser Art von Angriff erhält der Angreifer keine Informationen zurück, so dass er schwerer zu entdecken ist. Der Angreifer kann verschiedene Techniken anwenden, um Informationen über die Datenbank zu erlangen, z. B. durch Timing-Angriffe, um festzustellen, ob bestimmte Bedingungen wahr oder falsch sind.
+
+* **Out-of-band SQL-Injection**: Hierbei handelt es sich um eine Variante der blinden SQL-Injektion, bei der ein anderer Kanal zur Kommunikation mit dem Angreifer verwendet wird, z. B. DNS- oder HTTP-Anfragen an einen vom Angreifer kontrollierten Server.
+
+* **Second-Order SQL-Injection**: Bei dieser Art von Angriff wird der bösartige SQL-Code in der Datenbank gespeichert und zu einem späteren Zeitpunkt ausgeführt, etwa wenn ein anderer Benutzer auf die Daten zugreift. Dies kann die Erkennung und Entschärfung des Angriffs erschweren.
+
+Das Verständnis der verschiedenen Arten von SQL-Injection-Angriffen ist entscheidend für die Entwicklung wirksamer Gegenmaßnahmen und den Schutz von Webanwendungen vor diesen zunehmend verbreiteten und ausgefeilten Angriffen.
+
+## Was kann ich gegen SQL-Injections tun?
+
+Die beste Absicherung bietet eine Kombination von Best Practices, Tools und Techniken. Einige der wirksamsten Gegenmaßnahmen zur Verhinderung von SQL-Injections in einer Webanwendung sind:
+
+### Prepared Statements
+
+[Prepared Statements](https://de.wikipedia.org/wiki/Prepared_Statement) ist die einzig zuverlässige Art, um sich vor SQL Injections zu schützen.
+Vorausgesetzt, diese werden überall im Code angewendet.
+
+Prepared Statements sind SQL-Abfragen, die mit entsprechenden Parametern vorbereitet werden, aber erst zu einem späteren Zeitpunkt den eigentlichen Wert für den Parameter zugewiesen bekommen.
+Dabei wird der Wert des Parameters durch einen Placeholder markiert.
+Anbei ein Beispiel in Java:
+
+```java
+// Statement wird erzeugt
+PreparedStatement ps = connection.prepareStatement(
+    "SELECT user, password FROM tbl_user WHERE user = ?"
+);
+
+// Parameter werden übergeben
+ps.setString(1, username);
+
+// Statement wird ausgeführt.
+ResultSet rs = ps.executeQuery();
+
+```
+
+Doch was passiert da?
+
+1. Das Statement `SELECT user, password FROM tbl_user WHERE user = ?`an den Datenbank-Server gesendet und intern wird eine ID für das vorbereitete Statement gespeichert.
+2. Dem Parameter `?` (als Wert für das Feld `user`) die Zeichenkette aus der Variable `username` zugewiesen.
```

Comment:
der Client sendet die Zeichenkette und die ID der Abfrage an DB Server, der die eigentliche Abfrage erstellt und ausführt.

Reply:
Fixed in https://github.com/EngineeringKiosk/webpage/pull/231/commits/64d0d7bf20f909541307c9d2a1a8041ea92c3720